exporter/sematextexporter/writer_test.go

```diff
@@ -0,0 +1,221 @@
+		t.Run(testCase.name, func(t *testing.T) {
+			var httpRequests []*http.Request
+
+			mockHTTPService := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+				httpRequests = append(httpRequests, r)
+			}))
+			t.Cleanup(mockHTTPService.Close)
+
+			batch := &sematextHTTPWriterBatch{
+				sematextHTTPWriter: &sematextHTTPWriter{
+					encoderPool: sync.Pool{
+						New: func() any {
+							e := new(lineprotocol.Encoder)
+							e.SetLax(false)
+							e.SetPrecision(lineprotocol.Nanosecond)
+							return e
+						},
+					},
+					httpClient:      &http.Client{},
+					writeURL:        mockHTTPService.URL,
+					payloadMaxLines: testCase.payloadMaxLines,
+					payloadMaxBytes: testCase.payloadMaxBytes,
+					logger:          common.NoopLogger{},
+				},
+			}
+
+			err := batch.EnqueuePoint(context.Background(), "m", map[string]string{"k": "v"}, map[string]any{"f": int64(1)}, time.Unix(1, 0), 0)
+			require.NoError(t, err)
+			err = batch.EnqueuePoint(context.Background(), "m", map[string]string{"k": "v"}, map[string]any{"f": int64(2)}, time.Unix(2, 0), 0)
+			require.NoError(t, err)
+			err = batch.WriteBatch(context.Background())
+			require.NoError(t, err)
+
+			if testCase.expectMultipleRequests {
+				assert.Len(t, httpRequests, 2)
+			} else {
+				assert.Len(t, httpRequests, 1)
+			}
+		})
+	}
+}
+
+func Test_sematextHTTPWriterBatch_EnqueuePoint_emptyTagValue(t *testing.T) {
+	var recordedRequest *http.Request
+	var recordedRequestBody []byte
+	noopHTTPServer := httptest.NewServer(http.HandlerFunc(func(_ http.ResponseWriter, r *http.Request) {
+
+		if assert.Nil(t, recordedRequest) {
+			var err error
+			recordedRequest = r
+			recordedRequestBody, err = io.ReadAll(r.Body)
+			if err != nil {
+				fmt.Println(err.Error())
+			}
+		}
+	}))
+	t.Cleanup(noopHTTPServer.Close)
+	nowTime := time.Unix(1628605794, 318000000)
+
+	sematextWriter, err := newSematextHTTPWriter(
+		new(common.NoopLogger),
+		&Config{
+			ClientConfig: confighttp.ClientConfig{
+				Endpoint: noopHTTPServer.URL,
+			},
+			AppToken: "xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx",
+			Region:    "US",
+		},
+		componenttest.NewNopTelemetrySettings())
+	require.NoError(t, err)
+	sematextWriter.httpClient = noopHTTPServer.Client()
+	sematextWriterBatch := sematextWriter.NewBatch()
+
+	err = sematextWriterBatch.EnqueuePoint(
+		context.Background(),
+		"m",
+		map[string]string{"k": "v", "empty": ""},
+		map[string]any{"f": int64(1)},        
+		nowTime,                               
+		common.InfluxMetricValueTypeUntyped) 
+	require.NoError(t, err)
+
+	err = sematextWriterBatch.WriteBatch(context.Background())
+
+	require.NoError(t, err)
+
+	if assert.NotNil(t, recordedRequest) {
+		expected := "m,k=v,os.host=ADMINs-MacBook-Pro.local,token=xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx f=1i 1628605794318000000"
+		assert.Equal(t, expected, strings.TrimSpace(string(recordedRequestBody)))
+	}
+}
+
+func Test_composeWriteURL_doesNotPanic(t *testing.T) {
```

Comment:
consider going with something more Go-like for naming like `TestComposeWriteURLDoesNotPanic`

Reply:
Done